### PR TITLE
ENH: Allow multiline command blocks

### LINKED ISSUE #37: commandblock support for multi-line command execution
It would be great if Sphinx and/or our `commandblock` directive supported multiline commands:

```sh
$ qiime \
  tools \
  import \
  --help
```

### PR DESCRIPTION
**Pull request by jakereps**
Resolves #37

### CODE REVIEW

File: source/sphinx_directives/command_block.py [hunk 1]
```diff
@@ -185,6 +187,14 @@ def _get_output_links(self, output_paths, name):
                 content.append('')
         return content
 
+    def _parse_multiline_commands(self, previous, next):
+        if previous and previous[-1].endswith('\\'):
+            previous[-1].replace('\\', '')
+            previous[-1] += next
+        else:
+            previous.append(next)
```
Comment (ebolyen) [inline source/sphinx_directives/command_block.py]: not that it matters, but mutating the accumulator just feel so *wrong* to me...

File: source/sphinx_directives/command_block.py [hunk 2]
```diff
@@ -36,7 +37,8 @@ class CommandBlockDirective(docutils.parsers.rst.Directive):
 
     def run(self):
         self.assert_has_content()
-        commands = self.content
+        commands = functools.reduce(self._parse_multiline_commands,
```
Comment (ebolyen) [inline source/sphinx_directives/command_block.py]: This is a pretty slick solution

File: source/sphinx_directives/command_block.py [hunk 3]
```diff
@@ -185,6 +187,14 @@ def _get_output_links(self, output_paths, name):
                 content.append('')
         return content
 
+    def _parse_multiline_commands(self, previous, next):
+        if previous and previous[-1].endswith('\\'):
+            previous[-1].replace('\\', '')
```
Comment (ebolyen) [inline source/sphinx_directives/command_block.py]: It's unlikely to happen, but just doing a blanket replace might catch things that are escaped. I would refine this to only strip the last character.

**Comment by jakereps** [comment]
Thanks @ebolyen! Got those reducer changes in, and fixed the rendering that I forgot to check before submitting this. Passing the multiline parsed commands into the nodes was breaking the output, so switched it to pass `self.content` directly.

**Pull request merged**